webapp/playbooks/src/components/backstage/playbook_runs/playbook_run/rhs_participants.tsx

```diff
@@ -56,18 +56,18 @@ export const Participants = ({playbookRun, role, teamName}: Props) => {
     };
 
     useEffect(() => {
+        if (!playbookRun) {
+            return;
+        }
+
         const profiles = dispatch(getProfilesByIds(playbookRun.participant_ids));
 
         //@ts-ignore
         profiles.then(({data}: { data: UserProfile[] }) => {
-            // getProfilesByIds doesn't return current user profile, so add it when a user is participant
-            if (role === Role.Participant) {
-                data.push(myUser);
-            }
             data.sort(sortByUsername);
             setParticipantsProfiles(data || []);
         });
-    }, [dispatch, myUser, playbookRun.participant_ids, role]);
+    }, [dispatch, playbookRun, playbookRun.participant_ids, role]);
```

Comment:
I didn't track down exactly why, but `myUser` kept changing causing an infinite loop. In my test, the assertion that `getProfilesByIds` doesn't return current user profile was false, so I just simplified.

Reply:
Thanks for the deep dive @calebroseland and double checking my work here! Coming off PTO, I admit I don't have answers to some of what I was thinking, e.g. with the `!playbookRun` check. I've re-run tests without it and can't reproduce an issue.

I'm inclined to adopt all your proposes, except perhaps `.join('')`, as I'm not sure how to work around the linter warning of not explicitly depending on `participant_ids`.

I've pushed a proposed change -- let me know your thoughts?

Reply:
@lieut-data the change that you made will effectively work because of the sometimes-fetching `getMissingProfilesByIds` instead of the always-fetching `getProfilesByIds`.

While redundant network requests are mitigated, the root problem is still there—it will execute the effect whenever anything on the run changes. The `?.join('')` part was to base the effect dep off of the contents of the array, rather than the reference—but indeed not idiomatic.